### ISSUE TITLE
Wrong exception message about `toIPv4OrZero` function.

### ISSUE THREAD

**Issue by alexey-milovidov** [ClickHouse/ClickHouse#27490]
```
:) SELECT CAST('hello' AS IPv4)

SELECT CAST('hello', 'IPv4')

Query id: fd2510d2-6cbb-45c6-b3c6-a49bd03c4b9f


0 rows in set. Elapsed: 0.020 sec. 

Received exception from server (version 21.9.1):
Code: 6. DB::Exception: Received from localhost:9000. DB::Exception: Cannot parse string 'hello' as IPv4: syntax error at begin of string. Note: there are toIPv4OrZero and toIPv4OrNull functions, which returns zero/NULL instead of throwing exception.: While processing CAST('hello', 'IPv4'). (CANNOT_PARSE_TEXT)

:) SELECT toIPv4OrZero('hello')

SELECT toIPv4OrZero('hello')

Query id: f285c059-032a-4d11-95a9-283ef533427f


0 rows in set. Elapsed: 0.002 sec. 

Received exception from server (version 21.9.1):
Code: 46. DB::Exception: Received from localhost:9000. DB::Exception: Unknown function toIPv4OrZero. Maybe you meant: ['toInt8OrZero','toInt64OrZero']: While processing toIPv4OrZero('hello'). (UNKNOWN_FUNCTION)

:) SELECT toIPv4OrNull('hello')

SELECT toIPv4OrNull('hello')

Query id: 1ed6e303-bd6f-4631-b453-8db7efb21495


0 rows in set. Elapsed: 0.002 sec. 

Received exception from server (version 21.9.1):
Code: 46. DB::Exception: Received from localhost:9000. DB::Exception: Unknown function toIPv4OrNull. Maybe you meant: ['toInt8OrNull','toInt64OrNull']: While processing toIPv4OrNull('hello'). (UNKNOWN_FUNCTION)
```

**Comment by alexey-milovidov** [comment]
Tech debt from #3669.

**Comment by novikd** [comment]
Fixed in #29224
